### DISCUSSION TITLE
Create a list of all tailwind classes used in a project.

### DISCUSSION THREAD

**Discussion by mwawrusch** [tailwindlabs/tailwindcss#13515] [Ideas]
It would be beneficial to be able to extract a list of all tailwind classes *used* in a project for refactoring purposes. So for example I'd love to know the different variations of z-.. and then consolidate them manually.

THere is another discussion about extracting all possible classes, but that' not what I am looking for (https://github.com/tailwindlabs/tailwindcss/discussions/10379).

**Comment by limikael** [comment]
Yes this would be very useful! If someone would write such a tool they would be a hero!

**Comment by limikael** [comment]
Here is a start... It finds all string constants (e.g. anything surrounded by " ` '), but only the constants that includes a hyphen. It finds all my tailwind classes, but also a whole lot of other things. What is the actual algorithm that tailwind uses to find classes?

```
grep -o -E "['\`\"][^'\`\"]*-[^'\`\"]*['\`\"]" `find src -name "*.*js*"`
```

**Comment by Ingramz** [comment]
Some basic usage statistics could be useful as well to expose classes only referenced once in order to cut down on output size or to expose inconsistent styling used within the project.

**Comment by craigrileyuk** [comment]
https://mangle.icebreaker.top/en/patch

You only need to use the 'patch' part of the software, that'll give you a .json file with all detected classes.
